### ISSUE TITLE
3rd party root modules not finding by simple name

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#4417]
````
$ javac -cp "checkstyle-7.8-all.jar" com/puppycrawl/tools/checkstyle/TestRootModule.java

$ cat com/puppycrawl/tools/checkstyle/TestRootModule.java
package com.puppycrawl.tools.checkstyle;

public class TestRootModule extends Checker {
}

$ jar -cf MyCustom.jar com/puppycrawl/tools/checkstyle/TestRootModule.class

$ javac TestClass.java

$ cat TestClass.java
public class TestClass {
    void method() {
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="TestRootModule">
    <property name="charset" value="UTF-8"/>
</module>

$ java -classpath MyCustom.jar;checkstyle-7.8-all.jar com.puppycrawl.tools.checkstyle.Main -c TestConfig.xml TestClass.java
com.puppycrawl.tools.checkstyle.api.CheckstyleException: Unable to instantiate 'TestRootModule' class, it is also not possible to instantiate it as com.puppycrawl.tools.checkstyle..TestRootModule, TestRootModuleCheck, com.puppycrawl.tools.checkstyle..TestRootModuleCheck. Please recheck that class name is specified as canonical name or read how to configure short name usage http://checkstyle.sourceforge.net/config.html#Packages. Please also recheck that provided ClassLoader to Checker is configured correctly.
    at com.puppycrawl.tools.checkstyle.PackageObjectFactory.createModule(PackageObjectFactory.java:166)
    at com.puppycrawl.tools.checkstyle.Main.getRootModule(Main.java:449)
    at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:416)
    at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:359)
    at com.puppycrawl.tools.checkstyle.Main.main(Main.java:174)
Checkstyle ends with 1 errors.
````

Make key note to the packages it is trying to instantiate them by. They all have double periods. This is preventing custom root modules from being instantiated.

Checkstyle 7.7 doesn't have double periods and works fine. This was most likely broken by Issue #3607 .
````
$ java -classpath MyCustom.jar;checkstyle-7.7-all.jar com.puppycrawl.tools.checkstyle.Main -c TestConfig.xml TestClass.java
Starting audit...
Audit done.
````

[Our test didn't pick this up because it uses a full class path to load the module](https://github.com/checkstyle/checkstyle/blob/45d5f97b47f6b392ac0170914d1776deae2005af/src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java#L804).

As a side note, the list of packages it tries for the check is very limited, most likely because it doesn't even load the list from the `checkstyle_packages.xml` file.

**Comment by romani** [comment]
@rnveach , please share links to TestRootModule , goal of CLI usage in issue reporting is to reproduce issue easily and without problems. Right now your test case can not be reproduced easily, extra jar in `-cp ....` should be referenced.

**Comment by rnveach** [comment]
> please point to code of UT

Added to first post.

> please share links to TestRootModule 

Name of module was just example. This is why I used an exception error to show the problem.
[RootModule that I found error with is in my branch embeded into Checkstyle](https://github.com/rnveach/checkstyle/blob/issue_138/src/main/java/com/puppycrawl/tools/checkstyle/MultiThreadedChecker.java) so there is no extra JAR needed.
Exception is as it appears with released checkstyle which shows the double period problem.
I will edit the post for a dummy example.
